src/transformers/models/univnet/feature_extraction_univnet.py

```diff
@@ -0,0 +1,466 @@
+        max_length_s: int = 10,
+        fmin: float = 0.0,
+        fmax: Optional[float] = None,
+        mel_floor: float = 1e-9,
+        center: bool = False,
+        compression_factor: float = 1.0,
+        compression_clip_val: float = 1e-5,
+        normalize_min: float = -11.512925148010254,
+        normalize_max: float = 2.3143386840820312,
+        model_in_channels: int = 64,
+        pad_end_length: int = 10,
+        spectrogram_zero: float = -11.5129,
+        **kwargs,
+    ):
+        super().__init__(
+            feature_size=feature_size,
+            sampling_rate=sampling_rate,
+            padding_value=padding_value,
+            **kwargs,
+        )
+
+        self.do_normalize = do_normalize
+
+        self.num_mel_bins = num_mel_bins
+        self.hop_length = hop_length
+        self.win_length = win_length
+        self.win_function = win_function
+        self.filter_length = filter_length
+        self.fmin = fmin
+        if fmax is None:
+            # Follows the librosa.filters.mel implementation
+            fmax = float(sampling_rate) / 2
+        self.fmax = fmax
+        self.mel_floor = mel_floor
+
+        self.max_length_s = max_length_s
+        self.num_max_samples = max_length_s * sampling_rate
+
+        if self.filter_length is None:
+            self.n_fft = optimal_fft_length(self.win_length)
+        else:
+            self.n_fft = self.filter_length
+        self.n_freqs = (self.n_fft // 2) + 1
+
+        self.window = window_function(window_length=self.win_length, name=self.win_function, periodic=True)
+
+        self.mel_filters = mel_filter_bank(
+            num_frequency_bins=self.n_freqs,
+            num_mel_filters=self.num_mel_bins,
+            min_frequency=self.fmin,
+            max_frequency=self.fmax,
+            sampling_rate=self.sampling_rate,
+            norm="slaney",
+            mel_scale="slaney",
+        )
+
+        self.center = center
+        self.compression_factor = compression_factor
+        self.compression_clip_val = compression_clip_val
+        self.normalize_min = normalize_min
+        self.normalize_max = normalize_max
+        self.model_in_channels = model_in_channels
+        self.pad_end_length = pad_end_length
+        # TODO: Calculate spectrogram_zero? Should be a function of self.mel_floor I think
+        self.spectrogram_zero = spectrogram_zero
+
+    # Based on tacotron2.audio_processing.dynamic_range_compression
+    # https://github.com/NVIDIA/tacotron2/blob/master/audio_processing.py#L78
+    def dynamic_range_compression(self, waveform: np.ndarray):
+        return np.log(np.clip(waveform, a_min=self.compression_clip_val, a_max=None) * self.compression_factor)
+
+    # Based on tacotron2.audio_processing.dynamic_range_compression
+    # https://github.com/NVIDIA/tacotron2/blob/master/audio_processing.py#L87
+    def dynamic_range_decompression(self, waveform: np.ndarray):
+        return np.exp(waveform) / self.compression_factor
```

Comment:
See https://github.com/huggingface/transformers/pull/24799#discussion_r1326808316.

Reply:
I think this should be resolved since `dynamic_range_compression` has been moved into `mel_spectrogram` (see https://github.com/huggingface/transformers/pull/24799#discussion_r1344086684 and https://github.com/huggingface/transformers/pull/24799#discussion_r1346641935). Feel free to reopen if you feel there is still an issue :).